nl_server/embeddings.py

```diff
@@ -22,11 +24,60 @@
 from sentence_transformers.util import semantic_search
 import torch
 
+from nl_server import query_util
 import nl_server.gcs as gcs
+from server.lib.nl import utils
 
 TEMP_DIR = '/tmp/'
 MODEL_NAME = 'all-MiniLM-L6-v2'
 
+# A value higher than the highest score.
+_INIT_SCORE = 1.1
+
+# Scores below this are ignored.
+_SV_SCORE_THRESHOLD = 0.5
+
+# If the difference between successive scores exceeds this threshold, then SVs at
+# the lower score and below are ignored.
+_MULTI_SV_SCORE_DIFFERENTIAL = 0.05
+
+_NUM_CANDIDATES_PER_NSPLIT = 3
+
+
+# List of SV candidates, along with scores.
+@dataclass
+class VarCandidates:
+  # The below are sorted and parallel lists.
+  svs: List[str]
+  scores: List[float]
+  sv2sentences: Dict[str, List[str]]
```

Comment:
Might not worth a change for now, but why not having a list of {sv, score, sentences} struct?